src/gui/plugins/playback_scrubber/PlaybackScrubber.cc

```diff
@@ -0,0 +1,235 @@
+    percentage = 0;
+  if (percentage > 1.0)
+    percentage = 1.0;
+  return percentage;
+}
+
+//////////////////////////////////////////////////
+void PlaybackScrubber::Update(const UpdateInfo &_info,
+    EntityComponentManager &_ecm)
+{
+  auto startTime = this->dataPtr->startTime;
+  auto endTime = this->dataPtr->endTime;
+  auto totalDuration =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(
+        endTime - startTime).count();
+  if (totalDuration <= 0)
+  {
+    _ecm.Each<components::LogPlaybackStatistics>(
+      [this](const Entity &,
+           const components::LogPlaybackStatistics *_logStatComp)->bool
+      {
+        auto startSeconds =
+          _logStatComp->Data().start_time().sec();
+        auto startNanoseconds =
+          _logStatComp->Data().start_time().nsec();
+        auto endSeconds =
+          _logStatComp->Data().end_time().sec();
+        auto endNanoseconds =
+          _logStatComp->Data().end_time().nsec();
+        this->dataPtr->startTime =
+          math::secNsecToTimePoint(startSeconds, startNanoseconds);
+        this->dataPtr->endTime =
+          math::secNsecToTimePoint(endSeconds, endNanoseconds);
+        return true;
+      });
+  }
+
+  auto simTime = math::durationToSecNsec(_info.simTime);
+  this->dataPtr->currentTime =
+    math::secNsecToTimePoint(simTime.first, simTime.second);
+  this->dataPtr->progress = CalculateProgress();
+  this->dataPtr->paused = _info.paused;
+  this->newProgress();
+}
+
+/////////////////////////////////////////////////
+double PlaybackScrubber::Progress()
+{
+  return this->dataPtr->progress;
+}
+
+/////////////////////////////////////////////////
+QString PlaybackScrubber::StartTimeAsString()
+{
+  return QString::fromStdString(
+      math::timePointToString(this->dataPtr->startTime));
+}
+
+/////////////////////////////////////////////////
+QString PlaybackScrubber::EndTimeAsString()
+{
+  return QString::fromStdString(
+      math::timePointToString(this->dataPtr->endTime));
+}
+
+/////////////////////////////////////////////////
+QString PlaybackScrubber::CurrentTimeAsString()
+{
+  return QString::fromStdString(
+      math::timePointToString(this->dataPtr->currentTime));
+}
+
+/////////////////////////////////////////////////
+void PlaybackScrubber::OnTimeEntered(const QString &_time)
+{
+  std::string time = _time.toStdString();
+  std::chrono::system_clock::time_point enteredTime =
+    math::stringToTimePoint(time);
+  if (enteredTime == std::chrono::system_clock::from_time_t(-1))
+  {
+    ignwarn << "Invalid time entered. "
+      "The format is dd hh:mm:ss.nnn" << std::endl;
+    return;
+  }
+
+  // Check for time out of bounds
+  if (enteredTime < this->dataPtr->startTime)
+    enteredTime = this->dataPtr->startTime;
+  else if (enteredTime > this->dataPtr->endTime)
+    enteredTime = this->dataPtr->endTime;
+
+  auto pairTime = math::timePointToSecNsec(enteredTime);
+  const std::string topic = "/world/default/playback/control";
```

Comment:
We shouldn't hardcode the world name here, you should be able to get it from [ignition::gui::worldNames()](https://ignitionrobotics.org/api/gui/3.3/namespaceignition_1_1gui.html#a16c452c451b1405a336472b2ed6a9c01).

Reply:
So I'm not sure if this was a bug or not.  If I get the proper world name and use it, the playback callback doesn't work, but if I use `default` for the world name it works.

Reply:
Resolved e7a428e1e3fbd4c02f4f17a14817dcadf989b751